pootle/apps/pootle_fs/plugin.py

```diff
@@ -0,0 +1,344 @@
+        return pootle_state.get(self.__class__)
+
+    @lru_cache(maxsize=None)
+    def get_fs_path(self, pootle_path):
+        """
+        Reverse match an FS filepath from a ``Store`` using the project fsconf.
+
+        :param store: A ``Store`` object to get the FS filepath for
+        :returns: An filepath relative to the FS root.
+        """
+        return self.matcher.reverse_match(pootle_path)
+
+    def clear_repo(self):
+        if self.is_cloned:
+            shutil.rmtree(self.project.local_fs_path)
+
+    def create_store_fs(self, fs_path=None, pootle_path=None, store=None):
+        return self.store_fs_class.objects.create(
+            project=self.project,
+            pootle_path=pootle_path,
+            path=fs_path,
+            store=store)
+
+    def find_translations(self, fs_path=None, pootle_path=None):
+        """
+        Find translation file from the file system
+
+        :param force: Add conflicting translations.
+        :param fs_path: Path glob to filter translations matching FS path
+        :param pootle_path: Path glob to filter translations to add matching
+          ``pootle_path``
+        :yields pootle_path, fs_path:
+        """
+        return self.matcher.matches(fs_path, pootle_path)
+
+    def pull(self):
+        """
+        Pull the FS from external source if required.
+        """
+        raise NotImplementedError
+
+    def push(self, paths=None, message=None, response=None):
+        """
+        Push the FS to an external source if required.
+        """
+        raise NotImplementedError
+
+    def reload(self):
+        self.project.config.reload()
+        if "matcher" in self.__dict__:
+            del self.__dict__["matcher"]
+        if "pootle_user" in self.__dict__:
+            del self.__dict__["pootle_user"]
+
+    def response(self, state):
+        return self.response_class(state)
+
+    def state(self, fs_path=None, pootle_path=None):
+        """
+        Get a state object for showing current state of FS/Pootle
+
+        :return state: Where ``state`` is an instance of self.state_class
+        """
+        self.pull()
+        return self.state_class(
+            self, fs_path=fs_path, pootle_path=pootle_path)
+
+    @responds_to_state
+    def add(self, state, response,
+            fs_path=None, pootle_path=None, force=False):
+        """
+        Add translations from Pootle into the FS
+
+        If ``force``=``True`` is present it will also:
+        - add untracked conflicting files from Pootle
+        - mark tracked conflicting files to update from Pootle
+
+        :param force: Add conflicting translations.
+        :param fs_path: Path glob to filter translations matching FS path
+        :param pootle_path: Path glob to filter translations to add matching
+          ``pootle_path``
+        """
+        to_add = state["pootle_untracked"]
+        if force:
+            to_add = (
+                to_add
+                + state["conflict_untracked"]
```

Comment:
`+`? Which type is `state["conflict_untracked"]`? If boolean I suggest using `or` instead.